code/modules/clothing/clothing.dm

```diff
@@ -752,6 +753,18 @@
 		var/mob/M = src.loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/proc/trip_up(var/turf/walking, var/running)
+	if(!running)
+		if(footstep >= 2)
+			footstep = 0
+		else
+			footstep++
+	else
+		if(prob(25))
+			if(ismob(usr))
+				var/mob/M = usr
+				M.Weaken(2)
```

Comment:
`usr` in procs is not something we want to use, generally. It's unsafe, therefore we should ask ourselves here - how do we do this in a safer fashion?

The simpler solution is just to check the loc of the shoes - which should always be a mob, and if it is, trip 'em.

Also add a warning message so people know why they're tripping.

Reply:
Fixed